%&<latex>
\documentclass[letterpaper,12pt]{article}

\pdfpagewidth = 8.5in
\pdfpageheight = 11.0in
\usepackage[left=1in,right=1in,top=1in,bottom=1in]{geometry}
\usepackage{setspace}
\usepackage{xspace}
\usepackage{authblk}

\pagestyle{plain}
\pagenumbering{arabic}

\usepackage[round]{natbib}
\usepackage{hyperref}
% \hypersetup{pdfborder={0 0 0}, colorlinks=true, urlcolor=black, linkcolor=black, citecolor=black}
\hypersetup{pdfborder={0 0 0}, colorlinks=true, urlcolor=blue, linkcolor=blue, citecolor=blue}

\newcommand{\highLight}[1]{\textcolor{magenta}{\MakeUppercase{#1}}}
\newcommand{\ignore}[1]{}
\newcommand{\super}[1]{\ensuremath{^{\textrm{#1}}}}
\newcommand{\sub}[1]{\ensuremath{_{\textrm{#1}}}}
\newcommand{\dC}{\ensuremath{^\circ{\textrm{C}}}}
\newcommand{\booktitle}{\textit{Bayesian Phylogenetics}\xspace}
\newcommand{\booksubtitle}{\textit{Methods, algorithms, and applications}\xspace}
\newcommand{\bookfulltitle}{\textit{\booktitle: \booksubtitle}\xspace}
\newcommand{\editors}{Ming-Hui Chen, Lynn Kuo, and Paul O.\ Lewis\xspace}

\title{Book review: \bookfulltitle, edited by \editors}

\author[1,2]{Jamie R.\ Oaks\thanks{Corresponding author: \href{mailto:joaks@auburn.edu}{\tt joaks@auburn.edu}}}

\affil[1]{Department of Biology, University of Washington, Seattle, Washington 98195}
\affil[2]{Department of Biological Sciences, Auburn University, Auburn, Alabama 36849}

\date{\today}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\maketitle

\newpage
\doublespacing

\bookfulltitle, edited by \editors \citep{Chen2014}, is a collection of 13
papers contributed by leaders in the field of Bayesian phylogenetics.
A strong emphasis of the book is Bayesian model choice, with eight chapters
(Chapters 2--9) relevant to approximating marginal likelihoods,
six chapters (Chapters 2--6, 9) make this their main emphasis.

Given the amount of overlap among the chapters, an effort to better maintain
consistency of notation across the chapter would have made it easier to flip
back and forth between chapters.

What this book is not.
It is not a gentle introduction to Bayesian phylogenetics.
This is not the book for someone looking for a gentle introduction
to the field of Bayesian phylogenetics.
This book is for someone with a background in statistical phylogenetics
who is looking to get up to speed with the current state and future directions
of the field of Bayesian phylogenetics.
Rather it is collection of contributions from leaders in the field
represents the cutting-edge of the field

In chapter 2, Wang and Yang focus on priors in Bayesian phylogenetics.
They give a succinct introduction to objective versus subjective perspectives
of Bayesian priors, but maintain the focus of the chapter on evaluating
the sensitivity of the posterior to the prior.
They use a simple example of estimating the distance between between
two DNA sequences as a platform to introduce the likelihood function
and uniform, Jeffreys, and reference priors.

They do a great job of demonstrating the potential problems
with high dimension of the inference problem
with
using seemingly diffuse, independent priors on the length
of each branch in the tree.
They show the very strong (and often biologically implausible) prior
this places on the sum of these parameters (i.e., the length of the tree),
and demonstrate the influence this can have on the posterior by
analyzing an empirical dataset.

*   Coverage of Jeffreys and Reference priors is a bit cursory.
*   Very nice empirical demonstration of problem with i.i.d. branch length
    priors.



Given the strong focus of the book on Bayesian model choice, a meaningful
review of the book is not possible without a light (and very informal)
introduction to the currency of Bayesian model choice, the marginal likelihood,
and the challenges of computing it.
First, we need a brief (and informal) introduction to the challenges
of Bayesian model choice.

The currency of Bayesian model choice is the marginal likelihood: The
probability of the data under a model, averaged, with respect to the prior,
over the whole parameter space.
Due to the necessary summation or integration over the large number of parameters
in phylogenetic models
Given the large number of parameters in phylogenetic models that need
to be summed or integrated, this marginalized measure of model fit cannot
be computed analytically.
In fact, the beauty of MCMC is that it allows us to sample from the posterior
while avoiding ever having to calculate the marginal likelihood.
However, if we want to compare models, we can't avoid it.

The simplest numerical approximation to this nasty integral is to simply take
samples of model's parameters from the prior and turn the integral into a sum
of the likelihood of each sample (remembering to correct for the weights of the
samples to keep everything proper, which amounts to dividing by the number of
samples in this case).
This simplifies to the average of the likelihoods of the prior samples.
Almost as simple, if we had a sample of the parameters from the posterior
distribution, we can again sum the likelihood of each sample, but this time the
weight of each sample is not 1, but rather the ratio of the prior density to
the unnormalized posterior density.
This ends up simplifying to the harmonic mean (HM) of the likelihoods of the
posterior sample \citep{Newton1994}.
These are both importance-sampling approximations of the marginal likelihood,
and both suffer severely from the fact that the prior and posterior are often
very divergent from one another (with the latter often much more peaked than
the former).
As a result, a finite sample from the prior is likely to miss the
region of parameter space with high likelihood, and thus often
underestimate the marginal likelihood.
Similarly, a finite sample from the posterior is likely to have no samples
outside of the region of high posterior density, where the prior has a strong
downward ``pull'' on the average likelihood, and thus will often fail to
penalize for parameters, resulting in over estimates of the marginal
likelihood.

There are two ways to mitigate the problem caused by the strong difference
between the prior and posterior.
One is to take many small steps, sampling from a series of power posterior
distributions, along the path connecting the unnormalized posterior to the
prior (i.e., turn a giant leap into a bunch of small steps);
the path-sampling (PS) \citep{Lartillot2006} and stepping-stone (SS)
\citep{Xie2011} methods take this approach.
The other is to not use the prior as the reference distribution, but rather a
different proper probability distribution that is as similar as possible to the
posterior (i.e., turn a giant leap into a smaller leap); 
generalized harmonic mean (GHM) estimators \citep{Gelfand1994} can use
this approach.
The former strategy requires a lot of extra computation that is avoided by the
latter, but approaches that use the latter strategy tend to have greater error.
The most efficient numerical method for approximating the marginal likelihood
in a phylogenetic context to date is the generalized stepping-stone (GSS)
technique \citep{Fan2011}.
The GSS combines both strategies, sampling from a series of power
posteriors along the path connecting the unnormalized posterior to
an arbitrary proper reference distribution, the closer to the
posterior the better.


The GSS was restricted to estimating marginal likelihoods for a fixed tree
topology, because a reference distribution for trees was lacking.
In Chapter 5, Holder et al.\ introduce a reference distribution on trees that
is parameterized by a sample of trees from the posterior.
A distribution that samples trees in rough proportion to the posterior promises
to be broadly applicable beyond the context of the GSS method.
For example, it can easily be used for generalized versions of HM (GHM) and PS
(GPS) techniques.
Arima and Tardella (Chapter 3), allude to this by implementing fixed-tree GPS
and GHM methods.


Arima and Tardella (Chapter 3) explore the performance of the inflated density
ratio (IDR) approach for estimating marginal likelihoods of phylogenetic models
\citep{Arima2012}.
The IDR is a variant of the GHM estimator where the use of the prior in the HM
estimator is replaced by a perturbation of the unnormalized posterior density.
They compared the IDR and a version of the GHM estimator that relies on the
same reference distribution of GSS, to the HM, SS, GSS, PS, and GPS techniques,
using simulated and biological data.
They find that the GSS and GPS are the best-performing estimators of the
marginal likelihood.
The GHM and IDR methods both perform much better than the HM estimator, thus
providing viable options when the computational burden of PS and SS approaches
is not practical.

Baele and Lemey (Chapter 4) assess the ability of various methods to accurately
choose the true relaxed-clock model from simulated sequence data.
They compare HM estimators, AICM (which estimates the Akaike information criterion
from the likelihoods of posterior samples), and PS and SS techniques.
Furthermore they compare all of these model-choice approaches to the maximum
\emph{a posteriori} model estimated under Bayesian model averaging via Markov
chain Monte Carlo (MCMC).
They find that PS, SS, and MAP perform similarly and better than AICM
and HM estimators.
Baele and Lemey also use the SS approach to compare the fit of demographic
models to sequences from HIV viruses.
They find the marginal likelihood estimates from multiple demographic models
are too similar to be confident in the best model, given the variance in SS
estimates.

Prior to this book,
the IDR estimator was restricted to a fixed tree, because a perturbation of the
posterior distribution over trees was not available.
Wu et al. (Chapter 6) extend the IDR method to marginalize over trees.
They also prove the statistical consistency of the HM, GSS, and IDR estimators
when marginalizing over trees.
Using an empirical dataset, they also show that IDR performs much better than
the HM estimator, and provides a computationally efficient alternative to the
most accurate, and computationally demanding methods.

Palczewski and Beerli (Chapter 9) derive an identity showing that the overall
marginal likelihood over an arbitrary number of independent subsets of 
data (i.e., loci in population genetic models) can be calculated from
the marginal likelihoods calculated independently from each subset. 
This enables the marginal likelihood to be calculated from large-scale
datasets, by distributing the computation across many processors.

Beyond the model-choice focus of the book, there are several other chapters \ldots
Cheon and Liang (Chapter 7) and Bouchard-C\^{o}t\'{e} (Chapter 8) highlight numerical
integration algorithms that promise to complement the standard workhorse, MCMC,
for approximating posteriors of phylogenetic models.
After a very thorough review of numerical algorithms used in phylogenetics,
Cheon and Liang introduce a sequential stochastic approximation Monte Carlo (SSAMC) \citep{Liang2007,Cheon2008}
algorithm.
Using simulated data, they show that SSAMC can avoid getting trapped in local
regions of high posterior density, while exhibiting smaller per-iteration CPU
times than popular MCMC implementations.

Bouchard-C\^{o}t\'{e} (Chapter 8) provides a very intuitive treatment of
sequential Monte Carlo algorithms for approximating posteriors of trees.
Highlighting the "embarrassingly parallel" nature of SMC, he discusses
the computational gains SMC can offer via parallelization.
Bouchard-C\^{o}t\'{e} also discusses how the sequential nature of the
algorithm may allow it to accommodate richer models of sequence evolution
that can cripple MCMC.

Newly developing algorithms are discussed by X, Y, and Z,
which hold promise to increase the computational efficiency of Bayesian
phylogenetic inference, and make more biologically rich models of evolution
computationally feasible.

Kuhner (Chapter 10) gives an intuitive overview of the extreme challenges
posed by sampling the space of ancestral recombination graphs (ARG).
While the difficult nature of the problem has made progress on estimating
ARGs slow, she highlights opportunities for future progress, which is ever more
important, as more biologists realize that reticulate processes of evolution
often should not be ignored.

Palacios et al.\ (Chapter 11) reviews nonparametric Bayesian methods of
inferring the effective size of populations through time. By framing
the problem as estimating the intensity function of a point process,
they show that a rich literature of computational techniques
for point processes provides avenues for continuing to move
this field forward.
Using simulated data, they also show how increasing the number
of loci can lead to more precise estimates of the effective
population size through time.

Currently, models of sequence evolution that are used in phylogenetics are
largely restricted to models that make it computationally feasible to calculate
the likelihood via Felsenstein's pruning algorithm \citep{Felsenstein1981}.
To potentially circumvent this limitation, Hobolth and Thorne (Chapter 12)
propose Bayesian algorithms that can exploit the simpler likelihood form
of a fully observed Markov process.
Not only are such methods exciting due to their potential to make biologically
richer models of evolution computationally feasible, 
but also because they permit inferences of the detailed evolutionary history
of sequences along the tree.
% explore Markov models of sequence evolution that are conditioned upon
% the endpoints of a branch.
% endpoint-conditioned trajectories of DNA
% sequence evolution using Markov models.

Heath and Moore (Chapter 13) provide a very thorough review of Bayesian methods
for inferring ultrametric phylogenies (with branch lengths in units  of
relative or absolute time), without assuming a constant rate of nucleotide
substitution across the tree.



Many of the authors take advantage of the venue to write less formally to give
the reader a better intuition of the models and methods being described.

References are great resource.

One thing that is missing is consistent divulging of code. E.g., 
If I wanted to use the VIDR estimator of marginal likelihood (Chapter 6),
I cannot.

\bibliographystyle{sysbio}
\bibliography{references}

\end{document}

